js/graviex.js

```diff
@@ -0,0 +1,1058 @@
+        //     "created_at": "2022-02-04T09:28:09+03:00",
+        //     "volume": "0.141",
+        //     "remaining_volume": "0.141",
+        //     "executed_volume": "0.0",
+        //     "trades_count": 0,
+        //     "strategy": null
+        //   }
+        // ]
+        return this.parseOrders (response, market, since, limit);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    encodeParams (params) {
+        if ('orders' in params) {
+            const orders = params['orders'];
+            let query = this.urlencode (this.keysort (this.omit (params, 'orders')));
+            for (let i = 0; i < orders.length; i++) {
+                const order = orders[i];
+                const keys = Object.keys (order);
+                for (let k = 0; k < keys.length; k++) {
+                    const key = keys[k];
+                    const value = order[key];
+                    query += '&orders%5B%5D%5B' + key + '%5D=' + value.toString ();
+                }
+            }
+            return query;
+        }
+        return this.urlencode (this.keysort (params));
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + '/' + path;
+        const request = '/webapi/v3/' + path;
+        const query = this.omit (params, this.extractParams (path));
+        if (api === 'public' && path === 'tickers/{market}') {
+            url = this.urls['api'][api] + this.implodeParams (path, params);
+        } else if (api === 'public' && path !== 'tickers/{market}') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const nonce = this.nonce ().toString ();
+            const query = this.encodeParams (this.extend ({
+                'access_key': this.apiKey,
+                'tonce': nonce,
+            }, params));
+            const payload = method + '|' + request + '|' + query;
+            const signed = this.hmac (this.encode (payload), this.encode (this.secret), 'sha256');
+            const suffix = query + '&signature=' + signed;
+            if (method === 'GET') {
+                url += '?' + suffix;
+            } else {
+                body = suffix;
+                headers = { 'Content-Type': 'application/x-www-form-urlencoded' };
+            }
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    async request (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined, config = {}, context = {}) {
+        const response = await this.fetch2 (path, api, method, params, headers, body, config, context);
+        return response;
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        if (response === undefined) {
```

Comment:
![image](https://user-images.githubusercontent.com/43336371/153578407-ee3422b2-ca8b-4350-b3ef-fe021c09cc56.png)

You should try to understand whenever we got an errored response from the exchange, parse that error and match it with the adequate exception (following our error hierarchy)
In this case, you could extract the code '2002' and match it with the `invalidOrder` exception

Very rough example:

First, add the exceptions object to the exchange definition (top of the file)
```
            'exceptions': {
                'exact': {
                    '2002': InvalidOrder, //  {"error":{"code":2002,"message":"Failed to create order. Reason: Volume is too small"}}
```

Then here in the handleErrors:

```
const error = this.safeValue(response, 'errror);
const code = this.safeValue(error, 'code);
if (code !== undefined) {
            const feedback = this.id + ' ' + body;
            this.throwExactlyMatchedException (this.exceptions['exact'], code, feedback);
}
```

Reply:
@carlosmiei, Added the exception handling as you described above please check-in latest commit https://github.com/ccxt/ccxt/commit/bb2b0b2ef29c557eee9dad97fee9c010de932c9a